src/screenshot.cpp

```diff
@@ -831,6 +832,40 @@ bool MakeHeightmapScreenshot(const char *filename)
 	return sf->proc(filename, HeightmapCallback, nullptr, MapSizeX(), MapSizeY(), 8, palette);
 }
 
+static ScreenshotType _screenshot_type;
+
+/**
+ * Callback on the confirmation window for huge screenshots.
+ * @param w Window with viewport
+ * @param confirmed true on confirmation
+ */
+static void ScreenshotConfirmationCallback(Window *w, bool confirmed)
+{
+	if (confirmed) MakeScreenshot(_screenshot_type, nullptr);
+}
+
+/**
+ * Make a screenshot.
+ * Ask for confirmation if the screenshot will be huge.
+ * @param t Screenshot type: World, defaultzoom, heightmap or viewport screenshot
+ */
+void TakeScreenshot(ScreenshotType t)
+{
+	ViewPort vp;
+	SetupScreenshotViewport(t, &vp);
+	if ((uint64)vp.width * (uint64)vp.height > 8192 * 8192) {
+		/* Ask for confirmation */
+		_screenshot_type = t;
+		SetDParam(0, vp.width);
+		SetDParam(1, vp.height);
+		ShowQuery(STR_WARNING_SCREENSHOT_SIZE_CAPTION, STR_WARNING_SCREENSHOT_SIZE_MESSAGE, nullptr, ScreenshotConfirmationCallback);
+	}
+	else {
```

Comment:
Given I reviewed the previous changes for this file, I'm not sure how i missed this many style issues - 
`} else {`

Reply:
I probably shouldn't have missed that either. I'll fix it, of course; such is the bane of code maintenance :wink: